pyrobosim/pyrobosim/core/robot.py

```diff
@@ -138,6 +139,7 @@ def set_pose(self, pose):
         :type pose: :class:`pyrobosim.utils.pose.Pose`
         """
         self.dynamics.pose = pose
+        self.location = self.world.get_location_from_pose(pose)
```

Comment:
I'd like to avoid blanket adding this because this calculation is fairly expensive to do every time a pose is set, but I understand why this is helpful.

At the same time, the GUI (in `gui/world_canvas.py`) is doing this calculation separately so long as the actively selected robot is currently moving. So at the very least we should remove that duplicate logic from there if this function is already taking care of it.

Let me think about it a bit...

Reply:
If set_pose is called during normal runs then I agree.  We just need to make sure to call on the initial set of random initial pose and whenever the "random pose" button is pressed.

Reply:
I gave it an attempt in here: https://github.com/sea-bass/pyrobosim/pull/263 -- how does this work for you? It would subsume this PR, since I wasn't sure I could push to your fork.